helpers/calendar/analytics_helper.py

```diff
@@ -175,16 +176,37 @@ def get_room_statistics(self, number_of_events_in_room, all_details):
         return result
 
     @staticmethod
-    def get_total_bookings(self, query, start_date, end_date):
+    def get_total_bookings(instance, query, start_date, end_date, room_id=None):
         bookings = 0
-        rooms = CommonAnalytics.get_room_details(self, query)
+        rooms = CommonAnalytics.get_room_details(instance, query)
+        active_rooms = RoomModel.query.filter(RoomModel.state == "active")
         for room in rooms:
-            all_events = CommonAnalytics.get_all_events_in_a_room(
-                self, room["room_id"], start_date, end_date)
-            if all_events:
-                bookings += len(all_events)
+            if room_id:
+                exact_room = active_rooms.filter(
+                    RoomModel.id == room_id).first()
+                if not exact_room:
+                    raise GraphQLError("Room Id does not exist")
+                if room.get('calendar_id') == exact_room.calendar_id:
```

Comment:
I don't think this line is necessary here.

Reply:
@Bonifase this condition enables one to check for a specific calendar_Id whose booking count we need to be returned. so believe it is necessary. kindly shed some more light on the same if you think otherwise.

Reply:
thanx for point this out @Bonifase. this has been refactored.